src/components/course/data/hooks.jsx

```diff
@@ -186,14 +186,13 @@ export function useCourseRunWeeksToComplete(courseRun) {
   const [label, setLabel] = useState(undefined);
 
   useEffect(() => {
```

Comment:
@brobro10000 Unrelated to your specific changes, I'm curious to hear your two cents on whether these hooks like `useCourseRunWeeksToComplete ` to extract data from `courseRun` _need_ to use `useEffect` and/or `useState`.

I'm thinking there's probably ways to write these hooks without needing a `useEffect`. For example:

```jsx
export function useCourseRunWeeksToComplete(courseRun) {
  let weeksToComplete;
  let label;

  if (courseRun?.weeksToComplete >= 0) {
    weeksToComplete = courseRun.weeksToComplete;
    if (courseRun.weeksToComplete === 1) {
      label = 'week';
    } else {
      label = 'weeks';
    }
  }
  
  return [weeksToComplete, label];
}
```

Reply:
The benefit of moving away from `useEffect` for these basic hooks is that using the hook shouldn't trigger any component re-renders due to state changes.

Reply:
The refactor does not change the testing outcome, and I do agree a refactor of all the hooks in this file to functions as opposed to react state variable is necessary. 

Will defer as tech debt along with leaving an additional comment in this file as a TODO

Reply:
Sounds good! Should we add it as a row to the Tech Debt Tracker as a way to capture the tech debt in writing outside of code, too? https://2u-internal.atlassian.net/wiki/spaces/SOL/pages/15146378/Enterprise+Technical+and+Product+Debt+Tracker

Reply:
Yep, will add to it